src/EventStore.ClientAPI/EventStoreConnection.cs

```diff
@@ -107,9 +107,16 @@ public static IEventStoreConnection
 				}
 
 				if (scheme == "discover") {
-					var clusterSettings = new ClusterSettings(uri.Host, connectionSettings.MaxDiscoverAttempts,
-						uri.Port,
+					var port = uri.Port == -1 ? 2113 : uri.Port;
```

Comment:
Should we set the default gossip port to `Consts.DefaultClusterManagerExternalHttpPort;` here too? It seems that it would not be a backward-compatibility issue though since uri.Port=-1 was not handled before. But to maintain consistency, we may want to keep it the same.